torch/csrc/jit/codegen/cuda/tensor_view.cpp

```diff
@@ -427,6 +436,126 @@ TensorView* TensorView::computeAt(
   return this;
 }
 
+void TensorView::computeWith(
+    TensorView* consumer,
+    int consumer_pos,
+    bool best_effort) {
+  TORCH_INTERNAL_ASSERT(
+      !container()->isA<kir::Kernel>(),
+      "Function invalid for kernel container.");
+
+  if (isFusionInput()) {
+    return;
+  }
+
+  auto consumers = ir_utils::consumerTvsOf(this);
+  TORCH_CHECK(
+      std::find(consumers.begin(), consumers.end(), consumer) !=
+          consumers.end(),
+      "Not a direct consumer: ",
+      consumer->toString());
```

Comment:
Do we need to support the following use case?
```
T1, T2, T3 = Welford(T0)
T4 = set(T1)
T2->computeWith(T4, ...) // T4 is not a direct consumer of T2
```

Reply:
I don't think so as I don't see why we can't just do `T1->computeWith(T4, ...)` instead, unless that's too restrictive.